test_libs/pyspec/eth2spec/utils/ssz/ssz_typing.py

```diff
@@ -208,221 +138,115 @@ def __new__(cls, class_name, parents, attrs):
 
     def __getitem__(self, params):
         if not isinstance(params, tuple) or len(params) != 2:
-            raise Exception("Vector must be instantiated with two args: elem type and length")
-        o = self.__class__(self.__name__, (Vector,), {'elem_type': params[0], 'length': params[1]})
-        o._name = 'Vector'
+            raise Exception("List must be instantiated with two args: elem type and length")
+        o = self.__class__(self.__name__, (self,), {'elem_type': params[0], 'length': params[1]})
+        o._name = 'AbstractList'
         return o
 
-    def __subclasscheck__(self, sub):
-        return _is_vector_instance_of(self, sub)
-
-    def __instancecheck__(self, other):
-        return _is_vector_instance_of(self, other.__class__)
-
-    def __eq__(self, other):
-        return _is_equal_vector_type(self, other)
-
-    def __ne__(self, other):
-        return not _is_equal_vector_type(self, other)
-
-    def __hash__(self):
-        return hash(self.__class__)
-
+    def __instancecheck__(self, obj):
```

Comment:
not sure about these instance checks. If we're refactoring it again, it may be nice to find a better solution for the `issubclass(Vector[T, N], Vector)` and alike type checks.

Reply:
Back when we had multiple meta classes, I got it to actually work without relying on __name__ in this hacky way. Haven't figured out how to do it with a single meta class though.